packages/ui/src/theme/tokens/components/message.ts

```diff
@@ -0,0 +1,156 @@
+import { DesignTokenProperties, OutputVariantKey } from '../types/designToken';
+
+type MessageVariationTokens<OutputType> = DesignTokenProperties<
+  'backgroundColor' | 'color' | 'borderColor',
+  OutputType
+> & {
+  info?: MessageColorThemeTokens<OutputType>;
+  error?: MessageColorThemeTokens<OutputType>;
+  warning?: MessageColorThemeTokens<OutputType>;
+  success?: MessageColorThemeTokens<OutputType>;
+};
+
+type MessageColorThemeTokens<OutputType> = DesignTokenProperties<
+  'backgroundColor' | 'color' | 'borderColor',
+  OutputType
+>;
+
+type MessageIconTokens<OutputType> = DesignTokenProperties<'size', OutputType>;
+
+type MessageHeadingTokens<OutputType> = DesignTokenProperties<
+  'fontSize' | 'fontWeight',
+  OutputType
+>;
+
+type MessageDismissTokens<OutputType> = DesignTokenProperties<
+  'gap',
+  OutputType
+>;
+
+type MessageTokenKey =
+  | 'alignItems'
+  | 'justifyContent'
+  | 'color'
+  | 'backgroundColor'
+  | 'borderColor'
+  | 'borderRadius'
+  | 'borderStyle'
+  | 'borderWidth'
+  | 'lineHeight'
+  | 'paddingBlock'
+  | 'paddingInline';
+
+export type MessageTokens<OutputType extends OutputVariantKey> =
+  DesignTokenProperties<MessageTokenKey, OutputType> & {
+    icon?: MessageIconTokens<OutputType>;
```

Comment:
Channeling inner Caleb here: prefer inline type rather than defining a type above if you only use it in one place.

Reply:
Updated in latest to use some inline types for icon, heading, and dismiss